demo/src/app/components/typeahead/overview/sections/customize-option-in-input/template.html

```diff
@@ -0,0 +1,19 @@
+<p>
+  It is also possible to customize the display of the option inside the input field, using the property <code>inputFormatter</code> which expects a function taking the option item as input and returning a string as output.
+</p>
+
+<p>
+  There are two limitations though:
```

Comment:
Why all this? Just say again (as for popup formatting):

Typeahead might use complex objects as options, so it would need to transform them to strings to display in the input when user selects an item (say with the mouse).

If your typeahead uses strings there is nothing to do, otherwise you would need to provide the `[inputFormatter]` function.

Reply:
All this is because it is not obvious that the input formatting might mess with future searches (depending on how it formats).

Some implementations might change the input element value when the user is about to edit the search term, exactly to avoid this. This is not the case in out implementation.

Some implementations might use complex layers to be able to display rich content inside what is considered to be the input, thus allowing more representations than just a simple string. This is not the case in out implementation.

This is for me a good point to highlight. I can simplify the wording though.

Reply:
I will rewrite it (keeping the info but making it simpler)